File: core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSourceLocal.java
package com.kin.ecosystem.core.data.settings;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;

public class SettingsDataSourceLocal implements SettingsDataSource.Local {

	private static final String SETTINGS_PREF_NAME_FILE_KEY = "kinecosystem_settings_data_source";

	private static final String IS_BACKED_UP_KEY = "backed_up_key";

	private final SharedPreferences settingsDataSourceSharedPref;

	public SettingsDataSourceLocal(@NonNull final Context context) {
		this.settingsDataSourceSharedPref = context.getSharedPreferences(SETTINGS_PREF_NAME_FILE_KEY, Context.MODE_PRIVATE);
	}

	@Override
	public void setIsBackedUp(final String publicAddress, boolean isBackedUp) {
		settingsDataSourceSharedPref.edit().putBoolean(IS_BACKED_UP_KEY + publicAddress, isBackedUp).apply();
	}

	@Override
	public boolean isBackedUp(final String publicAddress) {
		return settingsDataSourceSharedPref.getBoolean(IS_BACKED_UP_KEY + publicAddress, false);
	}
}

File: core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSourceLocal.kt
package com.kin.ecosystem.core.data.settings

import android.content.Context
import android.content.SharedPreferences

class SettingsDataSourceLocal(context: Context) : SettingsDataSource.Local {

    private val settingsDataSourceSharedPref: SharedPreferences

    init {
        this.settingsDataSourceSharedPref = context.getSharedPreferences(SETTINGS_PREF_NAME_FILE_KEY, Context.MODE_PRIVATE)
    }

    override fun setIsBackedUp(publicAddress: String, isBackedUp: Boolean) {
        settingsDataSourceSharedPref.edit().putBoolean(IS_BACKED_UP_KEY + publicAddress, isBackedUp).apply()
    }

    override fun isBackedUp(publicAddress: String): Boolean {
        return settingsDataSourceSharedPref.getBoolean(IS_BACKED_UP_KEY + publicAddress, false)
    }

    override fun setSawOnboarding(kinUserId: String) {
        settingsDataSourceSharedPref.edit().putBoolean(kinUserId, true).apply()
    }

    override fun isSawOnboarding(kinUserId: String): Boolean {
        return settingsDataSourceSharedPref.getBoolean(kinUserId, false)
    }

    companion object {

        private const val SETTINGS_PREF_NAME_FILE_KEY = "kinecosystem_settings_data_source"

        private const val IS_BACKED_UP_KEY = "backed_up_key"
    }
}
